src/System.Net.HttpListener/tests/HttpListenerPrefixCollectionTests.cs

```diff
@@ -216,7 +216,7 @@ public void Add_PrefixAlreadyRegisteredAndStarted_ThrowsHttpListenerException()
 
         [Theory]
         [InlineData("http://microsoft.com/")]
-        [InlineData("http://[]/")]
+        //Managed implementation allows this. Is this a bad test? [InlineData("http://[]/")]
```

Comment:
Okay. Just a quick question then, as I know very little about the S.N stack and http.sys internals.
As of this PR, the argument validation for hosts is shared between windows and managed. This is great.
However, the Windows implementation then calls `HttpAddUrlToUrlGroup` which performs extra argument validation.
Obviously we can't do this with the managed implementation, as this is a Windows native API.
So what's the best way to perform validation in exactly the same way as that c++ API? Do you think something like `Uri.CheckHostName` would have similar (or exactly the same) behaviour?